examples/medplum-fhircast-demo/src/hooks/index.ts

```diff
@@ -8,6 +8,7 @@ export function useClientId(): string {
 
 export function usePrevious<T>(value: T): T | undefined {
```

Comment:
What is this doing?  Seems like it's just a passthrough?

Reply:
Haha, yeah this a silly one but it basically keeps the value of the last render because of how `useEffect` works. If I remember correctly because of how `ref.current` is evaluated before the `useEffect` runs, it returns the value from the last render while storing the current value for the next render.

Basically it's useful for preventing running logic in hooks when nothing actually changed (something equivalent to `shouldComponentUpdate` functionality from class components).

Not sure why this hook isn't built in